src/onnx/onnx.cpp

```diff
@@ -448,6 +448,56 @@ struct onnx_parser
         return ins;
     }
 
+    instruction_ref reflect_pad(const std::vector<int64_t>& pads, instruction_ref input)
+    {
+        size_t num_dims = pads.size() / 2;
+        std::vector<int> ldims(num_dims);
+        std::vector<int> rdims(num_dims);
+
+        std::vector<int64_t> axes(num_dims);
+        std::iota(axes.begin(), axes.end(), int64_t{0});
+
+        for(int i = 0; i < num_dims; i++)
+        {
+            ldims.at(i) = pads.at(i);
+            rdims.at(i) = pads.at(i + num_dims);
+        }
+
+        // iterate over dimensions, starting from lowest dimension
+        for(int i = num_dims - 1; i >= 0; i--)
+        {
+            int axis    = i;
+            auto lcount = ldims.at(i);
+            auto rcount = rdims.at(i);
+            if(lcount == 0 and rcount == 0) // no padding for current dim
+                continue;
+
+            // calculate starts and ends for each iteration since shape may change
+            std::vector<size_t> dims = input->get_shape().lens();
+            std::vector<int64_t> starts(axes.size(), 0);
+            std::vector<int64_t> ends(dims.begin(), dims.end());
+            std::vector<instruction_ref> slices;
+
+            for(int j = 0; j < lcount; j++)
+            {
+                auto start_idx = (j + 1) % dims.at(i);
+                starts.at(i)   = start_idx;
+                ends.at(i)     = start_idx + 1;
```

Comment:
`starts.at(i)`and `ends.at(i)` can be assigned to a pointer or iterator outside of the loop so it doesn't need to do a lookup each time.